### PR TITLE
3418 installed plugins

### PR DESCRIPTION
**Pull request by cakekoa**
# What does this PR do?

Fixes part of #3418.

Adds the Installed Plugins page.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by performing a plugin upgrade and uninstall in the UI
* [x] If applicable, add screenshots or log transcripts of the feature working
    > https://github.com/guardicore/monkey/assets/11077625/755739ee-07af-499a-b447-65bf0c9891d5

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx [hunk 1]
```diff
@@ -1,36 +1,109 @@
...
   const refreshInstalledPlugins = () => {
     authComponent.authFetch('/api/agent-plugins/installed/manifests', {}, true)
       .then(res => res.json())
       .then(plugins => {
-        setInstalledPlugins(plugins);
+        setAllInstalledPlugins(plugins);
       });
   };
 
+  const refreshNuberOfUpgradablePlugins = () => {
```
Comment (ordabach) [inline monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx:63]: typo

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    ];
+  }
+
+  const getUninstallAction = (row) => {
+    const pluginId = row.id;
+    if (pluginsInUninstallProcess.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={pluginId + 'uninstall'}
+          icon={<DownloadingIcon/>}
```
Comment (ordabach) [inline monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:139]: maybe this icon is better?
import AutoDeleteIcon from '@mui/icons-material/AutoDelete';

your decision

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  const onUninstallClick = (pluginId, pluginType, pluginName) => {
+    setPluginsInUninstallProcess((prevState) => {
+      return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+    });
+
+    uninstallPlugin(pluginType, pluginName).then(() => {
+      setSuccessfullyUninstalledPluginsIds((prevState) => {
+        return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+      });
+      refreshInstalledPlugins();
```
[the remark this answers is not in the document]
Comment (cakekoa) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:44]: Can we do this in a separate PR?

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+import DeleteIcon from '@mui/icons-material/Delete';
+import DownloadingIcon from '@mui/icons-material/Downloading';
+import DownloadDoneIcon from '@mui/icons-material/DownloadDone';
+import UpgradeIcon from '@mui/icons-material/Upgrade';
+import RemoveDoneIcon from '@mui/icons-material/RemoveDone';
+import BasePlugins from './BasePlugins';
+import AuthComponent from '../../AuthComponent';
+
+
+const InstalledPlugins = () => {
```
Comment (ordabach) [inline monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:14]: If there's an upgrade in process, we need to disable the uninstall button and vice versa.

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx (hunk 2 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:139]: Looks good to me

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (shreyamalviya) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:44]: > Can we do this in a separate PR?

+1

File: monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx [hunk 5]
```diff
@@ -1,36 +1,109 @@
...
 export const PluginsContext = createContext({});
 
 const authComponent = new AuthComponent({});
 
 export const PluginState = () => {
-
+  const [allPlugins, setAllPlugins] = useState([]);
+  const [allInstalledPlugins, setAllInstalledPlugins] = useState([]);
   const [availablePlugins, setAvailablePlugins] = useState([]);
   const [installedPlugins, setInstalledPlugins] = useState([]);
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx:14]: It's very hard to tell what's the difference between all of these. Ideally, we should define types. 
I don't see why we need to store the unparsed data.

File: monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx (hunk 4 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:15]: > If there isn't new version, the upgrade needs to be hidden.

Already handled, the upgrade button is disabled in that case.


> If there's an upgrade in process, we need to disable the uninstall button and vice versa.

Done

Comment (shreyamalviya) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:15]: > Although the lists should refresh, let's not rely on it:
> 
> If the uninstall process finishes successfully, we need to hide the upgrade icon (if there is any)
> If the upgrade process finishes successfully, we need to hide the upgrade icon

How would we do that without lists? That's how were tracking state?

Comment (ordabach) [reply monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx:15]: Not sure what you mean.
We have the lists living in the context.

File: monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx (hunk 5 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/contexts/plugins/PluginsContext.tsx:14]: PR will come

**Pull request merged**